### PR TITLE
Sc 1793  nucliadb metrics bytes fields and searches

### PR DESCRIPTION
**Pull request by sunbit**
- Extend audit to provide insight of how many fields have changed and it's file fields storage movements
- Fix Audit of new resources was being set to MODIFIED
- Fix storage error when overwriting a file field
- Fix delete of a field when it has nested data

### CODE REVIEW

File: nucliadb_ingest/nucliadb_ingest/orm/processor.py [hunk 1]
```diff
@@ -442,7 +589,7 @@ async def create_kb(
             raise e
 
         if not failed:
-            storage = await get_storage(service_name=SERVICE_NAME)
+            storage = await get_storage()
```
Comment (bloodbare) [inline nucliadb_ingest/nucliadb_ingest/orm/processor.py:592]: Why we remove service_name ?

File: nucliadb_ingest/nucliadb_ingest/tests/fixtures.py [hunk 2]
```diff
@@ -274,14 +284,17 @@ async def txn(redis_driver):
 
 @pytest.fixture(scope="function")
 async def cache(redis):
+
     url = f"redis://{redis[0]}:{redis[1]}"
     pubsub = RedisPubsub(url)
+    set_utility(Utility.PUBSUB, pubsub)
```
Comment (bloodbare) [inline nucliadb_ingest/nucliadb_ingest/tests/fixtures.py:290]: Could we check if already exist ?

File: nucliadb_utils/nucliadb_utils/storages/gcs.py [hunk 3]
```diff
@@ -130,7 +130,7 @@ async def copy(
                 )
             else:
                 data = await resp.json()
-                assert data["name"] == destination_uri
+                assert data["resource"]["name"] == destination_uri
```
Comment (bloodbare) [inline nucliadb_utils/nucliadb_utils/storages/gcs.py:133]: Are you sure?

**Review by bloodbare — CHANGES_REQUESTED** [review]

File: nucliadb_ingest/nucliadb_ingest/fields/base.py [hunk 4]
```diff
@@ -122,11 +122,15 @@ async def db_set_value(self, payload: Any):
...
-        await self.resource.txn.delete(
-            KB_RESOURCE_FIELD.format(
-                kbid=self.kbid, uuid=self.uuid, type=self.type, field=self.id
-            )
+        field_base_key = KB_RESOURCE_FIELD.format(
+            kbid=self.kbid, uuid=self.uuid, type=self.type, field=self.id
         )
+        # Make sure we explicitly delete the field and any nested key
+        keys_to_delete = []
+        async for key in self.resource.txn.keys(field_base_key):
```
[the remark this answers is not in the document]
Comment (sunbit) [reply nucliadb_ingest/nucliadb_ingest/fields/base.py:130]: I'll check, but the first result of the `.keys` includes the provided key, so a single key`.keys` call will emit the same key.

[the remark this answers is not in the document]
Comment (bloodbare) [reply nucliadb_ingest/nucliadb_ingest/fields/base.py:130]: just make sure it works with the three drivers

**Review by bloodbare — APPROVED** [review]

**Pull request merged**